libfs/fs.go

```diff
@@ -841,3 +841,29 @@ func (fs *FS) ToHTTPFileSystem(ctx context.Context) http.FileSystem {
 func (fs *FS) RootNode() libkbfs.Node {
 	return fs.root
 }
+
+type folderHandleChangeObserver func()
+
+func (folderHandleChangeObserver) LocalChange(
+	context.Context, libkbfs.Node, libkbfs.WriteRange) {
+}
+func (folderHandleChangeObserver) BatchChanges(
+	context.Context, []libkbfs.NodeChange, []libkbfs.NodeID) {
+}
+func (o folderHandleChangeObserver) TlfHandleChange(
+	context.Context, *libkbfs.TlfHandle) {
+	o()
+}
+
+// SubscribeToEndOfLife returns a context that will be canceled when this *FS
+// reaches end-of-life, meaning if user of this object caches it for long term
+// use, it should invalide this entry and create a new one using NewFS.
+func (fs *FS) SubscribeToEndOfLife() (ctx context.Context, err error) {
```

Comment:
I think this should return a channel that gets closed, rather than a context.  You're not using any other part of the context, right?

Reply:
No I'm not using any other part, but this allows me not to worry about closing the channel more than once without building an extra layer. I guess we could use `WaitGroup` here too, but I figured `Context` was more appropriate. WDYT?

Reply:
I don't think it makes sense as a context or a waitgroup, sorry.  Those make it more confusing I think.  What about something like:

```go
c := make(chan struct{})
once := &sync.Once{}
fhco := folderHandleChangeObserver(func() { once.Do(close(c)) })
```

Reply:
Sure; will do!